### PR TITLE
tstest/integration/vms: end-to-end UDP test

### PR DESCRIPTION
**Pull request by Xe**
This tests incoming and outgoing UDP traffic. It would test incoming UDP
traffic however our socks server doesn't seem to allow for connecting to
destinations over UDP. When the socks server gets that support the
incoming test should pass without issue.

Signed-off-by: Christine Dodrill <xe@tailscale.com>

### CODE REVIEW

**Comment by Xe** [comment]
test run: https://github.com/tailscale/tailscale/pull/2361/checks?check_run_id=3022172199

File: tstest/integration/vms/udp_tester.go
Comment (crawshaw) [inline tstest/integration/vms/udp_tester.go:23]: `nc -u` isn't portable enough?

**Review by crawshaw — APPROVED** [review]

File: tstest/integration/vms/udp_tester.go
Comment (Xe) [reply tstest/integration/vms/udp_tester.go:23]: that implies that netcat is on all of the base images and the `netcat` package in the repos is the same netcat. This is not the case and I don't want to have to pick apart the differences if I can avoid it.

Comment (Xe) [reply tstest/integration/vms/udp_tester.go:23]: I'll add a comment though

**Comment by crawshaw** [comment]
FYI: my general review process is "approve with comments" means if you follow the requests in the comments, go ahead and merge. If a comment has a ? on it, it's by its nature just a request for information, so no need to block merging on it. We haven't fully standardized this across the company, it's on my TODO list.

**Comment by Xe** [comment]
I'm aware, I just wanted to wait for CI to pass :)

**Pull request merged**
